cmd/gapir/cc/main.cpp

```diff
@@ -71,44 +73,42 @@ std::unique_ptr<ResourceInMemoryCache> createResourceProvider(
     }
 }
 
-void listenConnections(Connection* conn,
-                       const char* authToken,
-                       const char* cachePath,
-                       int idleTimeoutMs,
-                       MemoryManager* memoryManager,
-                       core::CrashHandler& crashHandler) {
-    ServerListener listener(conn, memoryManager->getSize());
-
-    std::unique_ptr<ResourceInMemoryCache> resourceProvider(
-            createResourceProvider(cachePath, memoryManager));
-
-    while (true) {
-        std::unique_ptr<ServerConnection> acceptedConn(
-                listener.acceptConnection(idleTimeoutMs, authToken));
-        if (!acceptedConn) {
-            GAPID_INFO("Shutting down");
-            break;
-        }
-        std::unique_ptr<CrashUploader> crash_uploader = std::unique_ptr<CrashUploader>(new CrashUploader(crashHandler, *acceptedConn));
+// Setup creates and starts a replay server at the given URI port. Returns the
+// created and started server.
+std::unique_ptr<Server> Setup(const char* uri, const char* authToken,
+                              const char* cachePath, int idleTimeoutSec,
+                              core::CrashHandler* crashHandler,
+                              MemoryManager* memMgr) {
```

Comment:
Add TODO
This memMgr is not thread-safe. The original replay server does not handle multiple client connections, so it was fine to have just a single memory manager.

But with gRPC, the server is able to handle multiple connections. This results into multiple independent replay streams using the same memory manager, it can result into crashes/wrong result.

So far for each GAPIR, as we always have only one GAPIS talking to it, it won't cause an issue. But need note for future.

Reply:
Added at https://github.com/google/gapid/pull/1861/commits/2ea8a8635f2614c9b57130dd36be447c9a1d876c#diff-4f71d0359f7b5b6da6691abe3a4599e2R78